usage-examples/how-to-exchange-encrypted-data.md

```diff
@@ -0,0 +1,150 @@
+`@alice`
+#### 1. Lookup address of the _@alice_ atServer
+- Establish TLS connection to root.atsign.org:64
+- Send `alice\n`
+- Expect response in form `<someDomain.example.com>:<port>`
+#### 2. Authenticate to _@alice_ atServer
+- Establish TLS connection to the address of the `@alice` atServer
+- do PKAM authentication
+  - Send `from:@alice`
+  - Expect response in form `data:<challengeText>`
+  - Create a base64-encoded SHA256 signature of `challengeText` using the client's PKAM private key
+    - Dart code example
+    ```dart
+      var key = RSAPrivateKey.fromString(privateKey);
+      var sha256signature = key.createSHA256Signature(utf8.encode(challengeText) as Uint8List);
+      var signature = base64Encode(sha256signature);
+    ```
+  - Send `pkam:<signature>`
+  - Expect response `data:success` (or `error:<errorMessage>` if the signature could not be verified by the atServer
+    using the corresponding PKAM public key)
+#### 3a. Fetch existing shared symmetric key if it has already been created,
+- Send `llookup:shared_key.bob@alice` // This is a copy of the symmetric key which is encrypted with `@alice`'s 
+  public encryption key
+- If response like `data:<base64EncodedEncryptedSharedKey>`
+  - Decode from bas64, then decrypt using our private encryption key. The result will be a symmetric key, 
+    base64-encoded. **NB** _Currently we use RSA asymmetric keypairs, and AES-256 symmetric keys. We will be 
+    supporting other ciphers and modes in future._
+  - base64-decode the symmetric key => $sharedAESKey
+- Else if response like `error:error:AT0015-key not found : @bob:shared_key@alice does not exist in keystore` we
+  need to create a symmetric key and share it wih `@bob`
+#### 3b. or create a new shared symmetric key
+- Create a new AES-256 symmetric key => $sharedAESKey and base64-encode it => $base64EncodedSharedAESKey
+- Save for our own use in future
+  - Encrypt $base64EncodedSharedAESKey with our public encryption key and base64-encode the result => 
+    $base64EncodedEncryptedForAliceSharedAESKey
+  - Store in the atServer
+    - Send `update:shared_key.bob@alice $base64EncodedEncryptedForAliceSharedAESKey` and handle the response
+- Share with `@bob`
+  - Fetch `@bob`'s public encryption key
+    - Send `plookup:publickey@bob`
+    - Successful response `data:<bobsBase64EncodedPublicKey>`
+  - Encrypt $base64EncodedSharedAESKey with bobsBase64EncodedPublicKey and base64-encode the result =>
+    $base64EncodedEncryptedForBobSharedAESKey
+  - Store in the atServer
+    - Send `update:ttr:86400:@bob:shared_key@alice $base64EncodedEncryptedForBobSharedAESKey` and handle the 
+      response
+#### 4. Encrypt some data
+- Encrypt your $data with $sharedAESKey and base64-encode the result => $base64EncodedEncryptedData
+  - We currently use the CTR (aka SIC) mode of the AES cipher, with PKCS7Padding. Java example:
```

Comment:
Maybe important to mention the initialization vector an array of 16 zeroes?

Reply:
I'd rather not, given the imminent release of the fix to use new IVs every time